src/v/pandaproxy/rest/handlers.cc

```diff
@@ -81,10 +84,19 @@ get_brokers(server::request_t rq, server::reply_t rp) {
     auto make_metadata_req = []() {
         return kafka::metadata_request{.list_all_topics = false};
     };
+    // Servicing the request must be handled on the
+    // same core where the client is held. So we capture
+    // everything in a handler and pass that to the cache.
+    auto handler = [user{rq.user},
+                    authn_method{rq.authn_method},
+                    make_metadata_req](kafka_client_cache& cache) mutable {
+        client_ptr client{cache.fetch_or_insert(user, authn_method)};
+        return client->dispatch(make_metadata_req).finally([client] {});
+    };
+
     return rq.service()
-      .client()
-      .local()
-      .dispatch(make_metadata_req)
+      .client_cache()
+      .invoke_on_cache(rq.user, std::move(handler))
```

Comment:
sorta looks like `invoke_on_cache` should take the handler by value. the return statement here on 97 returns immediately and `handler` disappears from the stack if `invoke_on_cache` doesn't capture it by value some where. i commented back on invoke_on_cache about this, so this comment just highlights it a bit more.